src/command_modules/azure-cli-container/azure/cli/command_modules/container/custom.py

```diff
@@ -248,20 +248,38 @@ def _build_identities_info(identities):
 def _create_update_msi_role_assignment(cmd, resource_group_name, cg_name, identity_principle_id,
                                        role_definition_id, identity_scope):
     from azure.cli.core.profiles import ResourceType, get_sdk
+    from msrestazure.azure_exceptions import CloudError
 
     RoleAssignmentCreateParameters = get_sdk(cmd.cli_ctx, ResourceType.MGMT_AUTHORIZATION,
                                              'RoleAssignmentCreateParameters',
                                              mod='models', operation_group='role_assignments')
 
-    assignment_client = get_auth_management_client(cmd.cli_ctx, identity_scope).role_assignments
+    assignments_client = get_auth_management_client(cmd.cli_ctx, identity_scope).role_assignments
 
     role_assignment_guid = str(_gen_guid())
 
-    create_params = RoleAssignmentCreateParameters(
+    parameters = RoleAssignmentCreateParameters(
         role_definition_id=role_definition_id,
         principal_id=identity_principle_id
     )
-    return assignment_client.create(identity_scope, role_assignment_guid, create_params, custom_headers=None)
+
+    logger.info("Creating an assignment with a role '%s' on the scope of '%s'", role_definition_id, identity_scope)
+    retry_times = 30
```

Comment:
Why/How was this number of retries decided?

Reply:
I used a similiar value to what is used for role assigment retries in the arm command module. This comes out to be about 2 minutes. I can make it less if you think that is too long.

Reply:
I'm assuming you mean this: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/arm.py#L982 in core?
You should be able to call that function from core to use in your command module.

Reply:
Then i'll change to just use that